receiver/dockerstatsreceiver/docker.go

```diff
@@ -194,6 +194,61 @@ func (dc *dockerClient) toStatsJSON(
 	return &statsJSON, nil
 }
 
+func (dc *dockerClient) ContainerEventLoop(ctx context.Context) {
+	filters := dfilters.NewArgs([]dfilters.KeyValuePair{
+		{Key: "type", Value: "container"},
+		{Key: "event", Value: "destroy"},
+		{Key: "event", Value: "die"},
+		{Key: "event", Value: "pause"},
+		{Key: "event", Value: "stop"},
+		{Key: "event", Value: "start"},
+		{Key: "event", Value: "unpause"},
+		{Key: "event", Value: "update"},
+	}...)
+	lastTime := time.Now()
+
+EVENT_LOOP:
+	for {
+		options := dtypes.EventsOptions{
+			Filters: filters,
+			Since:   lastTime.Format(time.RFC3339Nano),
+		}
+		eventCh, errCh := dc.client.Events(ctx, options)
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case event := <-eventCh:
+				switch event.Action {
+				case "destroy":
+					dc.logger.Debug("Docker container was destroyed:", zap.String("id", event.ID))
+					dc.removeContainer(event.ID)
+				default:
+					dc.logger.Debug(
+						"Docker container update:",
+						zap.String("id", event.ID),
+						zap.String("action", event.Action),
+					)
+
+					if container, ok := dc.inspectedContainerIsOfInterest(ctx, event.ID); ok {
+						dc.persistContainer(container)
+					}
+				}
+
+				lastTime = time.Unix(0, event.TimeNano)
+			case err := <-errCh:
+				// We are only interested when the context hasn't been canceled
+				if ctx.Err() == nil {
+					dc.logger.Error("Error watching docker container events", zap.Error(err))
```

Comment:
iirc from the cli event code it's from invalid json responses or general errors from the daemon, as long as we constrain to be errors unrelated to the parent context being closed.

Reply:
I am not sure I understand your reply.
If it is a transient error it would be nice to add a comment explaining it in the code to make it clear why we keep running the loop. It is not apparent looking at the code.

Reply:
If the docker daemon were to temporarily refuse/drop connections or restart it seems that recovering would be preferable to logging an error and halting most dynamic update functionality.  Otherwise only the containers that are running from the beginning of the collector lifetime will be monitored.  Is the added comment ok?